### ISSUE TITLE
[1.5.0] Transitive dependency suffix gets overwritten by the scala version of current subproject

### ISSUE THREAD

**Issue by OlegYch** [sbt/sbt#6439]
## steps

sbt 1.5.0

`build.sbt`
```
lazy val settings = List(
  scalaVersion := "2.13.5",
)
lazy val a = project
  .settings(
    settings,
    libraryDependencies ++= List(
      "org.typelevel" % "macro-compat_2.13.0-RC2" % "1.1.1",
    )
  )

lazy val b = project.dependsOn(a).settings(
  settings,
)
```

## problem

```
sbt:sbt-test101> update
[warn]
[warn]  Note: Unresolved dependencies path:
[error] stack trace is suppressed; run last b / update for the full output
[error] (b / update) sbt.librarymanagement.ResolveException: Error downloading org.typelevel:macro-compat_2.13.5:1.1.1[error]   Not found
[error]   Not found
[error]   not found: C:\Users\OlegYch\.ivy2\local\org.typelevel\macro-compat_2.13.5\1.1.1\ivys\ivy.xml
[error]   not found: https://repo1.maven.org/maven2/org/typelevel/macro-compat_2.13.5/1.1.1/macro-compat_2.13.5-1.1.1.pom
[error] Total time: 0 s, completed Apr 9, 2021 3:22:06 PM
```
## expectation

dependency is not affected by current scala version

**Comment by eed3si9n** [comment]
@OlegYch Thanks for the report! That is a surprising behavior.

**Comment by eed3si9n** [comment]
Assuming this is a 1.5.0 regression, then likely something happened in https://github.com/sbt/librarymanagement/compare/v1.4.3...v1.5.0?

**Comment by ithinkicancode** [comment]
Hi, any progress on this issue?

My project doesn't have a direct dependency on `macro-compat` but one of the dependencies must have `macro-compat` because I'm getting the same error in sbt 1.5.x.

**Comment by eed3si9n** [comment]
I haven't looked into it yet.

In addition to looking into LM changes, it might be worth switching Coursier version to an older one and see if that would change anything.

**Comment by eed3si9n** [comment]
It works on sbt 1.4.9, and downgrading Coursier to 2.0.6 didn't make a difference as far as I can tell.
So it has to be related to something we did in lm or sbt/sbt for Scala 3?

**Comment by eed3si9n** [comment]
Never mind. I tried downgrading Coursier again, and I think downgrading lm-coursier-shaded to 2.0.7 or 2.0.6 fixed it.

```
$ cat project/build.properties
sbt.version=1.5.4-SNAPSHOT

$ sbt
[info] welcome to sbt 1.5.4-SNAPSHOT (AdoptOpenJDK Java 1.8.0_232)
[info] loading settings for project global-plugins from idea.sbt,pgp.sbt,plugins.sbt,vimquit.sbt ...
[info] loading global plugins from /Users/eed3si9n/.sbt/1.0/plugins
[info] loading project definition from /private/tmp/sbt-6439/project
[info] loading settings for project sbt-6439 from build.sbt ...
[info] set current project to sbt-6439 (in build file:/private/tmp/sbt-6439/)
[info] sbt server started at local:///Users/eed3si9n/.sbt/1.0/server/b05a7428022a9623d447/sock
[info] started sbt server
sbt:sbt-6439> clean
[success] Total time: 0 s, completed Jun 5, 2021 5:41:03 PM
sbt:sbt-6439> b/update
[success] Total time: 1 s, completed Jun 5, 2021 5:41:06 PM
```

**Comment by eed3si9n** [comment]
Hopefully fixed in #6725

**Comment by SiliconMeeple** [comment]
In the meantime, given that `1.5.6` is necessary for guarding against log4shell, is a good workaround for this something like `ThisBuild / useCoursier := false` ?
